File: libraries/analytics_debugger/src/main/java/com/tokopedia/analyticsdebugger/debugger/PerformanceLogger.java
package com.tokopedia.analyticsdebugger.debugger;

import com.tokopedia.analyticsdebugger.debugger.domain.model.PerformanceLogModel;

public interface PerformanceLogger {
    void save(PerformanceLogModel performanceLogModel);

    void wipe();

    void openActivity();

    void enableNotification(boolean status);

    boolean isNotificationEnabled();
}

File: libraries/analytics_debugger/src/main/java/com/tokopedia/analyticsdebugger/debugger/PerformanceLogger.kt
package com.tokopedia.analyticsdebugger.debugger

interface PerformanceLogger {

    val isNotificationEnabled: Boolean

    val isAutoLogFileEnabled: Boolean
    fun save(traceName: String,
             startTime: Long,
             endTime: Long,
             attributes: Map<String, String>,
             metrics: Map<String, Long>)

    fun wipe()

    fun openActivity()

    fun enableNotification(status: Boolean)

    fun enableAutoLogFile(status: Boolean)
}
